bemani/backend/iidx/rootage.py

```diff
@@ -0,0 +1,1299 @@
+                        ourindex = i
+                        break
+                if ourindex is None:
+                    raise Exception('Cannot find our own score after saving to DB!')
+                start = ourindex - 4
+                end = ourindex + 4
+                if start < 0:
+                    start = 0
+                if end >= len(all_scores):
+                    end = len(all_scores) - 1
+                relevant_scores = all_scores[start:(end + 1)]
+
+                record_num = start + 1
+                for score in relevant_scores:
+                    profile = all_players[score[0]]
+
+                    data = Node.void('data')
+                    data.set_attribute('iidx_id', str(profile.get_int('extid')))
+                    data.set_attribute('name', profile.get_str('name'))
+
+                    machine_name = ''
+                    if 'shop_location' in profile:
+                        shop_id = profile.get_int('shop_location')
+                        machine = self.get_machine_by_id(shop_id)
+                        if machine is not None:
+                            machine_name = machine.name
+                    data.set_attribute('opname', machine_name)
+                    data.set_attribute('rnum', str(record_num))
+                    data.set_attribute('score', str(score[1].points))
+                    data.set_attribute('clflg', str(self.db_to_game_status(score[1].data.get_int('clear_status'))))
+                    data.set_attribute('pid', str(profile.get_int('pid')))
+
+                    data.set_attribute('sgrade', str(
+                        self.db_to_game_rank(profile.get_int(self.DAN_RANKING_SINGLE, -1), self.s().GAME_CLTYPE_SINGLE),
+                    ))
+                    data.set_attribute('dgrade', str(
+                        self.db_to_game_rank(profile.get_int(self.DAN_RANKING_DOUBLE, -1), self.s().GAME_CLTYPE_DOUBLE),
+                    ))
+                    # achieve?
+
+                    qpro = profile.get_dict('qpro')
+                    data.set_attribute('head', str(qpro.get_int('head')))
+                    data.set_attribute('hair', str(qpro.get_int('hair')))
+                    data.set_attribute('face', str(qpro.get_int('face')))
+                    data.set_attribute('body', str(qpro.get_int('body')))
+                    data.set_attribute('hand', str(qpro.get_int('hand')))
+                    data.set_attribute('myFlg', '1' if score[0] == userid else '0')
+
+                    ranklist.add_child(data)
+                    record_num = record_num + 1
+            
+            # player_compe?
+
+        return root
+
+
+    # 0x1803EA8D0
+    def handle_IIDX26shop_request(self, request: Node) -> Node:
+        method = request.attribute('method')
+        root = Node.void('IIDX26shop')
+
+        # 0x18067D020
+        if method == 'sentinfo':
+            pass # ignore
+
+        # 0x18067D0B0
+        if method == 'getname':

```

Comment:
What happened to 'savename'?

Reply:
The callback function that decompiled doesn't contains the `savename` field, what was its functionality previously?